userspace/libsinsp/test/plugins/routines.cpp

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <driver/ppm_events_public.h>
+#include "sample_table.h"
+#include "test_plugins.h"
+
+struct plugin_state
+{
+    std::string lasterr;
+    ss_plugin_owner_t *owner;
+    ss_plugin_routine_vtable routine_vtable;
+
+    uint8_t step = 1;
+    bool flag = true;
```

Comment:
```suggestion
    std::atomic<bool> flag{true};
```
This needs to be atomic too, otherwise the test can go in infinite loop.